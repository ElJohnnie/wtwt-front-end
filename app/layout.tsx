import type { Metadata } from 'next';
import { Inter } from 'next/font/google';

const inter = Inter({ subsets: ['latin'] });

export const metadata: Metadata = {
  title: 'What to watch tonight?',
  description:
    'Um sistema de recomendação de filmes baseado em Machine Learning. Usar algoritmos de filtragem colaborativa ou baseados em conteúdo para recomendar filmes aos usuários baseado em gosto, vontades e humor',
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="pt-br">
      <body className={inter.className}>{children}</body>
    </html>
  );
}
