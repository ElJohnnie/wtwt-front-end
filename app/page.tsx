'use client';
import React, { useContext, useEffect } from 'react';
import { AppProvider, AppContext } from '../contexts/AppContext';
import Steps from '../components/core/Steps';
import Question from '../components/texts/Question';
import ButtonAnswer from '../components/buttons/ButtonAnswer';
import Description from '../components/texts/Description';
import StepsButton from '../components/buttons/StepsButton';
import { RoutesUrls } from '../utils/enums/routesUrl';
import { useNavigation } from '../hooks/useNavigation';
useNavigation;

const Home: React.FC = () => {
  const {
    steps,
    currentStep,
    questions,
    answeredSteps,
    answers,
    nextStep,
    prevStep,
    answerQuestion,
    showResult,
  } = useContext(AppContext);

  const { replace } = useNavigation();

  useEffect(() => {
    if (showResult) {
      replace(RoutesUrls.RESULT);
    }
  }, [showResult, replace]);

  return (
    <main className="flex min-h-screen flex-col items-center p-12">
      <div className="flex justify-between flex-row items-center my-4">
        {steps.current.map((step, index) => (
          <Steps
            key={index}
            number={step.number}
            hasRange={step.hasRange}
            isInactive={index > currentStep}
            asComplete={answeredSteps[index]}
          />
        ))}
      </div>
      <div className="flex items-center my-12">
        <Question text={questions[currentStep].question} />
      </div>
      <div className="my-4 grid grid-cols-2 gap-4">
        {questions[currentStep].answers.map((answer, index) => (
          <ButtonAnswer
            key={index}
            text={answer.text}
            selectedAnswer={answers[currentStep]}
            callback={answerQuestion}
          />
        ))}
      </div>
      <div className="flex items-center my-12">
        <Description text={questions[currentStep].description} />
      </div>
      <div className="flex flex-row items-center justify-center w-6/12 space-x-4">
        <div className={`${currentStep <= 0 ? 'hidden' : ''}`}>
          <StepsButton
            icon={
              <svg
                xmlns="http://www.w3.org/2000/svg"
                fill="none"
                viewBox="0 0 24 24"
                strokeWidth={1.5}
                stroke="currentColor"
                className="size-6"
                style={{ transform: 'scale(-1, 1)' }}
              >
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  d="M13.5 4.5L21 12m0 0l-7.5 7.5M21 12H3"
                />
              </svg>
            }
            callback={currentStep > 0 ? prevStep : undefined}
          />
        </div>
        <div className={`${!answeredSteps[currentStep] ? 'hidden' : ''}`}>
          <StepsButton
            icon={
              <svg
                xmlns="http://www.w3.org/2000/svg"
                fill="none"
                viewBox="0 0 24 24"
                strokeWidth={1.5}
                stroke="currentColor"
                className="size-6"
              >
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  d="M13.5 4.5L21 12m0 0l-7.5 7.5M21 12H3"
                />
              </svg>
            }
            callback={
              currentStep < steps.current.length - 1 ? nextStep : undefined
            }
          />
        </div>
      </div>
    </main>
  );
};

const HomePage: React.FC = () => {
  return (
    <AppProvider>
      <Home />
    </AppProvider>
  );
};

export default HomePage;
