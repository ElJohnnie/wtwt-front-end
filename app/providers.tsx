'use client';
import { AppProvider } from '../contexts/AppContext';
import { ThemeProvider } from '@mui/material/styles';
import theme from '../config/theme';

export function Providers({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <ThemeProvider theme={theme}>
      <AppProvider>{children}</AppProvider>
    </ThemeProvider>
  );
}
