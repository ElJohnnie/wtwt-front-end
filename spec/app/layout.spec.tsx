import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import RootLayout from '../../app/layout';
import Home from '../../app/page';
import { metadata } from '../../app/layout';

describe('RootLayout', () => {
  it('Renderizando o layout raiz', () => {
    const { container } = render(
      <RootLayout>
        <Home />
      </RootLayout>,
    );
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('Verifica a estrutura HTML e a classe de fonte', () => {
    const { getByText, container } = render(
      <RootLayout>
        <div>Teste de Conteúdo</div>
      </RootLayout>,
    );
    const htmlElement = container.querySelector('html');
    expect(htmlElement).toBeInTheDocument();
    expect(htmlElement).toHaveAttribute('lang', 'en');

    const bodyElement = container.querySelector('body');
    expect(bodyElement).toBeInTheDocument();
    expect(bodyElement).toHaveClass('className');
    expect(bodyElement).toContainElement(getByText('Teste de Conteúdo'));
  });

  it('Verifica os metadados', () => {
    expect(metadata.title).toBe('Create Next App');
    expect(metadata.description).toBe('Generated by create next app');
  });
});
