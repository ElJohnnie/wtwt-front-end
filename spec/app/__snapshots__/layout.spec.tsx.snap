// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout Renderizando o layout raiz 1`] = `
<div>
  <html
    lang="pt-br"
  >
    <body
      class="className"
    >
      <main
        class="flex min-h-screen flex-col items-center p-12"
      >
        <div
          class="flex justify-between flex-row items-center my-4"
        >
          <div
            class="flex flex-row justify-center items-center"
          >
            <div
              class="size-10 md:size-16 xl:size-18 rounded-full  
          inline-flex items-center justify-center  
          text-white text-[24px] font-bold bg-pink-600"
            >
              1
            </div>
            <div
              class="w-8 lg:w-24 md:w-16 bg-gray-200 rounded-full invisible md:visible mx-0 h-2.5 md:mx-4 dark:bg-gray-700"
            >
              <div
                class="bg-pink-600 h-2.5 rounded-full"
                style="width: 50%;"
              />
            </div>
          </div>
          <div
            class="flex flex-row justify-center items-center"
          >
            <div
              class="size-10 md:size-16 xl:size-18 rounded-full  
          inline-flex items-center justify-center  
          text-white text-[24px] font-bold bg-pink-400"
            >
              2
            </div>
            <div
              class="w-8 lg:w-24 md:w-16 bg-gray-200 rounded-full invisible md:visible mx-0 h-2.5 md:mx-4 dark:bg-gray-700"
            >
              <div
                class="bg-pink-600 h-2.5 rounded-full"
                style="width: 0%;"
              />
            </div>
          </div>
          <div
            class="flex flex-row justify-center items-center"
          >
            <div
              class="size-10 md:size-16 xl:size-18 rounded-full  
          inline-flex items-center justify-center  
          text-white text-[24px] font-bold bg-pink-400"
            >
              3
            </div>
            <div
              class="w-8 lg:w-24 md:w-16 bg-gray-200 rounded-full invisible md:visible mx-0 h-2.5 md:mx-4 dark:bg-gray-700"
            >
              <div
                class="bg-pink-600 h-2.5 rounded-full"
                style="width: 0%;"
              />
            </div>
          </div>
          <div
            class="flex flex-row justify-center items-center"
          >
            <div
              class="size-10 md:size-16 xl:size-18 rounded-full  
          inline-flex items-center justify-center  
          text-white text-[24px] font-bold bg-pink-400"
            >
              4
            </div>
            <div
              class="w-8 lg:w-24 md:w-16 bg-gray-200 rounded-full invisible md:visible mx-0 h-2.5 md:mx-4 dark:bg-gray-700"
            >
              <div
                class="bg-pink-600 h-2.5 rounded-full"
                style="width: 0%;"
              />
            </div>
          </div>
          <div
            class="flex flex-row justify-center items-center"
          >
            <div
              class="size-10 md:size-16 xl:size-18 rounded-full  
          inline-flex items-center justify-center  
          text-white text-[24px] font-bold bg-pink-400"
            >
              5
            </div>
          </div>
        </div>
        <div
          class="flex items-center my-12"
        >
          <div
            class="py-2 px-2 md:px-12 rounded-full font-bold text-pink-600 text-center text-2xl md:text-4xl"
          >
            Como você está se sentindo agora?
          </div>
        </div>
        <div
          class="my-4 grid grid-cols-2 gap-4"
        >
          <button
            class="focus:outline-none focus:ring-4 font-medium text-xl px-8 md:px-12 py-4 lg:px-24 lg:py-5 text-center mb-2 hover:bg-pink-800  rounded-full text-white bg-pink-600"
          >
            Feliz
          </button>
          <button
            class="focus:outline-none focus:ring-4 font-medium text-xl px-8 md:px-12 py-4 lg:px-24 lg:py-5 text-center mb-2 hover:bg-pink-800  rounded-full text-white bg-pink-600"
          >
            Triste
          </button>
          <button
            class="focus:outline-none focus:ring-4 font-medium text-xl px-8 md:px-12 py-4 lg:px-24 lg:py-5 text-center mb-2 hover:bg-pink-800  rounded-full text-white bg-pink-600"
          >
            Ansioso
          </button>
          <button
            class="focus:outline-none focus:ring-4 font-medium text-xl px-8 md:px-12 py-4 lg:px-24 lg:py-5 text-center mb-2 hover:bg-pink-800  rounded-full text-white bg-pink-600"
          >
            Bem animado
          </button>
          <button
            class="focus:outline-none focus:ring-4 font-medium text-xl px-8 md:px-12 py-4 lg:px-24 lg:py-5 text-center mb-2 hover:bg-pink-800  rounded-full text-white bg-pink-600"
          >
            Entediado
          </button>
        </div>
        <div
          class="flex items-center my-12"
        >
          <div
            class="text-center text-pink-600"
          >
            teste 123
          </div>
        </div>
        <div
          class="flex flex-row items-center justify-center w-6/12 space-x-4"
        >
          <div
            class="hidden"
          >
            <div
              class="w-full flex flex-col justify-center items-center"
            >
              <button
                class="w-24 h-24 rounded-full   inline-flex items-center justify-center   text-white text-[24px] font-bold bg-pink-600 hover:bg-pink-800"
              >
                <svg
                  class="size-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  style="transform: scale(-1, 1);"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.5 4.5L21 12m0 0l-7.5 7.5M21 12H3"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
              <p
                class="text-[18px] mt-1 text-pink-600"
              />
            </div>
          </div>
          <div
            class="hidden"
          >
            <div
              class="w-full flex flex-col justify-center items-center"
            >
              <button
                class="w-24 h-24 rounded-full   inline-flex items-center justify-center   text-white text-[24px] font-bold bg-pink-600 hover:bg-pink-800"
              >
                <svg
                  class="size-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.5 4.5L21 12m0 0l-7.5 7.5M21 12H3"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
              <p
                class="text-[18px] mt-1 text-pink-600"
              />
            </div>
          </div>
        </div>
      </main>
    </body>
  </html>
</div>
`;
